lib/signalfx/signalflow/client.rb

```diff
@@ -0,0 +1,58 @@
+require 'thread'
+
+require_relative "./websocket"
+
+# A SignalFlow client that supports swappable .
+class SignalFlowClient
+  def initialize(api_token, api_endpoint, stream_endpoint)
+    @transport = SignalFlowWebsocketTransport.new(api_token, stream_endpoint)
+  end
+
+  # Start a computation and attach to its output. If using WebSockets (the
+  # default), the channel name is handled internally so you do not need to
+  # supply it.
+  #
+  # See https://developers.signalfx.com/reference#section-execute-a-computation
+  #
+  # @option options [Fixnum] :start
+  # @option options [Fixnum] :stop
+  # @option options [Fixnum] :resolution
+  # @option options [Fixnum] :max_delay
+  # @option options [Boolean] :persistent
+  #
+  # @return [Computation] A {Computation} instance with an active channel
+  def execute(program, **options)
+    @transport.execute(program, **options)
+  end
+  
+  # Start and attach to a computation that tells how many times a detector
+  # would have fired in a time range between `start` and `stop`.
+  #
+  # See https://developers.signalfx.com/v2/reference#signalflowpreflight
+  #
+  # @param start [Fixnum]
+  # @param stop [Fixnum]
+  # @option options [Fixnum] :max_delay
+  #
+  # @return [Computation] A {Computation} instance with an active channel
+  def preflight(program, start, stop, **options, &block)
+    @transport.preflight(program, start, stop, **options, &block)
+  end
+
+  # Start a computation without attaching to it
+  #
+  # This is currently not very useful because you cannot attach to the
+  # computation to see any output.  It also will not return the handle id.
```

Comment:
Can you make it return the handle ID?

Reply:
It seems a lot more natural to return an unattached Computation object that the user can call `attach` and `stop` on.  I was having trouble getting a response from the server telling me the job handle though, but I'll try it again.  `start` is active on the prod endpoint already right?

Reply:
Yes, `start` works and returns the computation handle.